python/sports-league-scheduling/src/sports_league_scheduling/domain.py

```diff
@@ -0,0 +1,113 @@
+from timefold.solver import SolverStatus
+from timefold.solver.domain import (planning_entity, planning_solution, PlanningId, PlanningVariable,
+                                    PlanningEntityCollectionProperty,
+                                    ProblemFactCollectionProperty, ValueRangeProvider,
+                                    PlanningScore)
+from timefold.solver.score import HardSoftScore
+from typing import Dict, List, Any, Annotated
+
+from .json_serialization import *
+
+
+class Team(JsonDomainBase):
+    id: str
+    name: str
+    distance_to_team: Annotated[Dict[str, int],
+                                DistanceToTeamValidator,
+                                Field(default_factory=dict)]
+
+
+    def __eq__(self, other):
+        if self is other:  # Check if both references are the same
+            return True
+        if not isinstance(other, Team):  # Ensure the other object is a Team instance
+            return False
+        return self.id == other.id  # Compare IDs for equality
+
+    def __hash__(self):
+        return hash(self.id)
+
+    def __str__(self):
+        return self.id
+
+    def __repr__(self):
+        return f'Team({self.id}, {self.name}, {self.distance_to_team})'
+
+
+class Round(JsonDomainBase):
+    index: Annotated[int, PlanningId]
+    # Rounds scheduled on weekends and holidays. It's common for classic matches to be scheduled on weekends or holidays.
+    weekend_or_holiday: bool
```

Comment:
I would annotate this attribute with `Field(default=False)` so you wouldn't need to specify it every time in the test.

```suggestion
    weekend_or_holiday: Annotated[bool, Field(default=False)]
```